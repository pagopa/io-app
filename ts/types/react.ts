import { Component, ComponentType, StatelessComponent } from "react";
import {
  NavigationInjectedProps,
  NavigationScreenProps
} from "react-navigation";

/**
 * Evaluates to the Props type of a React component
 */
export type ComponentProps<C> = C extends StatelessComponent<infer P1>
  ? P1
  : C extends Component<infer P2>
    ? P2
    : C extends ComponentType<infer P3> ? P3 : never;

/**
 * Infers the type of the navigation params of a component
 */
export type InferNavigationParams<
  C,
  P = ComponentProps<C>
> = P extends NavigationInjectedProps<infer N>
  ? N
  : P extends NavigationScreenProps<infer N1> ? N1 : never;
