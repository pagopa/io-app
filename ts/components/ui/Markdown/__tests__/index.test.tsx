import * as React from "react";
import * as renderer from "react-test-renderer";

import { Markdown } from "..";

const mokedDispatch = jest.fn();

describe("Markdown", () => {
  describe("heading", () => {
    it("should render heading1 correctly", () => {
      const tree = renderer
        .create(
          <Markdown dispatch={mokedDispatch}># A simple heading1</Markdown>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should render heading2 correctly", () => {
      const tree = renderer
        .create(
          <Markdown dispatch={mokedDispatch}>## A simple heading2</Markdown>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should render heading3 correctly", () => {
      const tree = renderer
        .create(
          <Markdown dispatch={mokedDispatch}>### A simple heading3</Markdown>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("paragraph", () => {
    it("should render paragraph correctly", () => {
      const tree = renderer
        .create(
          <Markdown dispatch={mokedDispatch}>A simple paragraph.</Markdown>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("list", () => {
    it("should render unordered list correctly", () => {
      const tree = renderer
        .create(
          <Markdown
            dispatch={mokedDispatch}
          >{`A simple unordered list:\n* Vue\n* React\n* Angular`}</Markdown>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should render ordered list correctly", () => {
      const tree = renderer
        .create(
          <Markdown
            dispatch={mokedDispatch}
          >{`A simple ordered list:\n1. Vue\n2. React\n3. Angular`}</Markdown>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
