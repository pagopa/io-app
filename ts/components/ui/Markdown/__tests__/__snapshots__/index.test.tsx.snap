// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown heading should render heading1 correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontSize": 27,
        "lineHeight": 32,
      }
    }
  >
    A simple heading1
  </Text>
</View>
`;

exports[`Markdown heading should render heading2 correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontSize": 24,
        "lineHeight": 27,
      }
    }
  >
    A simple heading2
  </Text>
</View>
`;

exports[`Markdown heading should render heading3 correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontSize": 21,
        "lineHeight": 22,
      }
    }
  >
    A simple heading3
  </Text>
</View>
`;

exports[`Markdown list should render ordered list correctly 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      A simple ordered list
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      :
1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      . Vue
2
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      . React
3
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      . Angular
    </Text>
  </View>
</View>
`;

exports[`Markdown list should render unordered list correctly 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      A simple unordered list
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      :

    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      * Vue

    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      * React

    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      * Angular
    </Text>
  </View>
</View>
`;

exports[`Markdown paragraph should render paragraph correctly 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      A simple paragraph
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      .
    </Text>
  </View>
</View>
`;
